src/Workspaces/Core/Portable/Shared/Utilities/ProducerConsumer.cs

```diff
@@ -202,16 +255,16 @@ internal static class ProducerConsumer<TItem>
             () => channel.Writer.TryComplete(new OperationCanceledException(cancellationToken)));
 #endif
 
-        await Task.WhenAll(
-            ProduceItemsAndWriteToChannelAsync(),
-            ReadFromChannelAndConsumeItemsAsync()).ConfigureAwait(false);
+        var writeTask = ProduceItemsAndWriteToChannelAsync();
+        var readTask = ReadFromChannelAndConsumeItemsAsync();
+        await Task.WhenAll(writeTask, readTask).ConfigureAwait(false);
```

Comment:
>await Task.WhenAll(writeTask, readTask).ConfigureAwait(false); [](http://example.com/codeflow?start=8&length=62)

nit: as we're at the bowels, I feel like not allocating the array here is simple and worthwhile. Instead of doing the WhenAll, just awaiting writeTask will work the same without allocating the array, right?

Reply:
tecahnically we don't even need to await 'writeTask'.  we could just return 'readTask'.  but i like the clarity of this WhenAll :)